pinot-query-runtime/src/main/java/org/apache/pinot/query/mailbox/StringMailboxIdentifier.java

```diff
@@ -0,0 +1,108 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.query.mailbox;
+
+import com.google.common.base.Joiner;
+import com.google.common.base.Preconditions;
+
+
+public class StringMailboxIdentifier implements MailboxIdentifier {
+  private static final Joiner JOINER = Joiner.on(':');
+
+  private final String _mailboxIdString;
+  private final String _jobId;
+  private final String _partitionKey;
+  private final String _fromHost;
+  private final int _fromPort;
+  private final String _toHost;
+  private final int _toPort;
+
+  public StringMailboxIdentifier(String jobId, String partitionKey, String fromHost, int fromPort, String toHost,
+      int toPort) {
+    _jobId = jobId;
+    _partitionKey = partitionKey;
+    _fromHost = fromHost;
+    _fromPort = fromPort;
+    _toHost = toHost;
+    _toPort = toPort;
+    _mailboxIdString = JOINER.join(_jobId, _partitionKey, _fromHost, _fromPort, _toHost, _toPort);
+  }
+
+  public StringMailboxIdentifier(String mailboxId) {
+    _mailboxIdString = mailboxId;
+    String[] splits = mailboxId.split(":");
+    Preconditions.checkState(splits.length == 6);
+    _jobId = splits[0];
+    _partitionKey = splits[1];
```

Comment:
technically partitionKey shouldn't be part of the mailboxID. it should be a stateless function used by mailbox nodes to convert keySelector result (e.g. `uuid` to one of the group of receiving channels `fromHost:fromPort:toHost:toPort`)

Reply:
address this in follow up PR